components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/workflow/ApplicationCreationWSWorkflowExecutor.java

```diff
@@ -43,119 +43,92 @@
-			                         + "        <wor:applicationCallbackUrl>$3</wor:applicationCallbackUrl>\n"
-			                         + "        <wor:applicationDescription>$4</wor:applicationDescription>\n"	
-			                         + "        <wor:tenantDomain>$5</wor:tenantDomain>\n"	
-			                         + "        <wor:userName>$6</wor:userName>\n"
-			                         + "        <wor:workflowExternalRef>$7</wor:workflowExternalRef>\n"
-			                         + "        <wor:callBackURL>$8</wor:callBackURL>\n"
-			                         + "      </wor:ApplicationApprovalWorkFlowProcessRequest>";
-
-			ApplicationWorkflowDTO appWorkFlowDTO = (ApplicationWorkflowDTO) workflowDTO;
-			Application application = appWorkFlowDTO.getApplication();			
-			String callBackURL = appWorkFlowDTO.getCallbackUrl();			
-
-			
-			payload = payload.replace("$1", application.getName());		
-			payload = payload.replace("$2", application.getTier());
-			payload = payload.replace("$3", application.getCallbackUrl());
-			payload = payload.replace("$4", application.getDescription());
-			payload = payload.replace("$5", appWorkFlowDTO.getTenantDomain());	
-			payload = payload.replace("$6", appWorkFlowDTO.getUserName());
-			payload = payload.replace("$7", appWorkFlowDTO.getExternalWorkflowReference());
-			payload = payload.replace("$8", callBackURL != null ? callBackURL : "?");
-
-			client.fireAndForget(AXIOMUtil.stringToOM(payload));	
-			
-		} catch (AxisFault axisFault) {
-			log.error("Error sending out message", axisFault);
-			throw new WorkflowException("Error sending out message", axisFault);
-		} catch (XMLStreamException e) {
-			log.error("Error converting String to OMElement", e);
-			throw new WorkflowException("Error converting String to OMElement", e);
-		}
-
-	}
-
-	/**
-	 * Complete the external process status.
-	 * Based on the workflow , we will update the status column of the
-	 * Application table
-	 * 
-	 * @param workFlowDTO
-	 */
-	@Override
-	public void complete(WorkflowDTO workFlowDTO) throws WorkflowException {
+
+    private String serviceEndpoint;
+
+    private String username;
+
+    private String password;
+
+    private String contentType;
+
+    private static final Log log = LogFactory.getLog(ApplicationCreationWSWorkflowExecutor.class);
+
+    @Override
+    public String getWorkflowType() {
+        return WorkflowConstants.WF_TYPE_AM_APPLICATION_CREATION;
+    }
+
+    @Override
+    public void execute(WorkflowDTO workflowDTO) throws WorkflowException {
+
+        if (log.isDebugEnabled()) {
+            log.info("Executing Application creation Workflow..");
+        }
+        super.execute(workflowDTO);
+        try {
+            String action = WorkflowConstants.CREATE_APPLICATION_WS_ACTION;
+            ServiceClient client = getClient(action);
+
+            String payload =
+                    "<wor:ApplicationApprovalWorkFlowProcessRequest xmlns:wor=\"http://workflow.application.apimgt" +

```

Comment:
Hard-coding message payloads is a bad practice. Use an appropriate xml building object model instead.